PR: https://github.com/odoo/odoo/pull/282884

From: 549e3cdbd98614c7471d5da54660916ed721473f
From: svs-odoo
Date: 2026-08-28 20:53:36

Breaking data model changes scores: del:1 + add:0, change matches:
-    _inherit = "stock.picking.type"

Total Changes: 38

[MOV] stock,*: own file for picking type

*: stock_delivery

This commit creates its own file for the `stock.picking.type` model
which won't squat the 'stock.picking` model's file any more !
It's so good to have its own home ❤️

(More seriously, `stock_picking.py` is a big file and `StockPickingType`
has more than 500 lines for itself, so both models deserve to have
separate files, plus it's error prone to have them grouped in one file
because we can be working in a `StockPickingType` method while thinking
we're in `StockPicking`.)

Do the same in `stock_delivery`, the file wasn't as big but confusion
still can happen and that way we follow the coding guideline anyway
[^1].

task-6154300

[^1] "Each inherited model should be in its own file to help
understanding of impacted models."
https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#file-naming

Part-of: odoo/odoo#282884
Related: odoo/enterprise#128175
Related: odoo/upgrade#11054
Signed-off-by: William Henrotin (whe) <whe@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock_delivery/models/__init__.py
+++ b/addons/stock_delivery/models/__init__.py
@@ -8,5 +8,6 @@ from . import sale_order
 from . import stock_move
 from . import stock_package_type
 from . import stock_picking
+from . import stock_picking_type
 from . import stock_package
 from . import ir_actions_report

--- a/addons/stock_delivery/models/stock_picking.py
+++ b/addons/stock_delivery/models/stock_picking.py
@@ -4,7 +4,7 @@ from datetime import date
 from markupsafe import Markup
 import json
 
-from odoo import _, api, fields, models, SUPERUSER_ID
+from odoo import _, api, fields, models
 from odoo.exceptions import UserError
 from odoo.addons.web.controllers.utils import clean_action
 
@@ -267,19 +267,3 @@ class StockPicking(models.Model):
             clean_action(action, self.env)
             report_actions.append(action)
         return report_actions + super()._get_autoprint_report_actions()
-
-
-class StockPickingType(models.Model):
-    _inherit = "stock.picking.type"
-
-    auto_print_carrier_labels = fields.Boolean(
-        "Auto Print Carrier Labels",
-        help="Automatically print the carrier labels of the picking when they are created.",
-    )
-    auto_print_export_documents = fields.Boolean(
-        "Auto Print Export Documents",
-        help=(
-            "Automatically print the export documents of the picking when they are created. "
-            "Availability of export documents depends on the carrier and the destination."
-        ),
-    )

--- a/None
+++ b/addons/stock_delivery/models/stock_picking_type.py
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class StockPickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    auto_print_carrier_labels = fields.Boolean(
+        "Auto Print Carrier Labels",
+        help="Automatically print the carrier labels of the picking when they are created.",
+    )
+    auto_print_export_documents = fields.Boolean(
+        "Auto Print Export Documents",
+        help=(
+            "Automatically print the export documents of the picking when they are created. "
+            "Availability of export documents depends on the carrier and the destination."
+        ),
+    )
